### PR TITLE
Dye Fixes and Improvements

### PR DESCRIPTION
**Pull request by ProTriforcer**
- When you show uses for lapis, bonemeal, cocoa or ink sacs, you won't see 41 JEI pages of recipes making things blue, unless you craft them into a dye first. Inspired by 1.14 dyes.
This breaks some autocrafts, like concrete and other dyed things, but imo reducing JEI clutter is worth it
- Change elevator and slime block redyeing recipes to use dye oredicts instead of just vanilla items 
- Add the "dye" tag to all dyes to fix satchel redyeing

### CODE REVIEW

**Comment by IntegerLimit** [comment]
@tracer4b LGTM. Can you test it in game, and check balancing, etc.

Also, what do you think of this change?

File: overrides/scripts/undyeing.zs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+<ore:dye>.remove(<minecraft:dye:0>);
+recipes.addShapeless(<metaitem:dye.black>, [<minecraft:dye:0>]);
+
+<ore:dyeBrown>.remove(<minecraft:dye:3>);
+<ore:dye>.remove(<minecraft:dye:3>);
+recipes.addShapeless(<metaitem:dye.brown>, [<minecraft:dye:3>]);
+
+<ore:dyeBlue>.remove(<minecraft:dye:4>);
+<ore:dye>.remove(<minecraft:dye:4>);
+recipes.addShapeless(<metaitem:dye.blue>, [<minecraft:dye:4>]);
```
Comment (tracer4b) [inline overrides/scripts/undyeing.zs:12]: Should add conversions for lazurite, sodalite, and metal mixture dust to their respective dyes as well

**Review by tracer4b — CHANGES_REQUESTED** [review]

**Comment by ProTriforcer** [comment]
That should be all of them. Disregard the crushe _dYe_ llowLimonite
image

**Review by IntegerLimit — APPROVED** [review]

**Review by tracer4b — APPROVED** [review]

**Comment by Testure** [comment]
Please stop existing

**Comment by IntegerLimit** [comment]
As in... revert the PR, or?

**Comment by Testure** [comment]
Yes, revert back to sanity

**Comment by eutro** [comment]
Gregtech players when they need to perform one (1) additional craft to dye things in exchange for JEI no longer being incomprehensible for items that happen to be usable for dye.

**Comment by IntegerLimit** [comment]
No, we will not return to insanity. You shall perform the craft!

**Pull request merged**